v22.1/vault-db-secrets-tutorial.md

```diff
@@ -0,0 +1,472 @@
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+vault login $VAULT_CLIENT_TOKEN
+~~~
+
+~~~txt
+Success! You are now authenticated. The token information displayed below
+is already stored in the token helper. You do NOT need to run "vault login"
+again. Future Vault requests will automatically use this token.
+~~~
+
+### Confirm the limited permissions of the role
+
+To confirm that you have assumed a role with limited permissions, try listing the current leases on the `crdb-role`, as we did [previously](#). You should see a permissions error.
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+vault list sys/leases/lookup/database/creds/crdb-role/
+~~~
+
+~~~txt
+Error listing sys/leases/lookup/database/creds/crdb-role: Error making API request.
+
+Namespace: admin/
+URL: GET https://roach-test-vault.vault.bfb2290a-670b-4a10-bedf-5aab18e84d69.aws.hashicorp.cloud:8200/v1/sys/leases/lookup/database/creds/crdb-role?list=true
+Code: 403. Errors:
+
+* 1 error occurred:
+  * permission denied
+~~~
+
+### Pull the CockroachDB credentials
+
+The only thing this policy *does* have permission to do is pull credentials for the CockroachDB cluster. Let's do that.
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+vault read database/creds/crdb-role
+~~~
+
+~~~txt
+Key                Value
+---                -----
+lease_id           database/creds/crdb-role/V3T4UVxeQ9RYsJAk3jZF1Dhl.iMXtW
+lease_duration     1h
+lease_renewable    true
+password           FlOo0p7jMTXjT27hlZZ-H
+username           v-token-crdb-rol-thfLPlFwex0k9Op0P8qA-1653528652
+~~~
+  
+### Connect to the CockroachDB cluster with your Vault-generated dynamic secret credentials
+
+List all the tables in database `defaultdb` to confirm you can connect to your CockroachDB cluster.
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+USER_NAME=v-token-crdb-rol-thfLPlFwex0k9Op0P8qA-1653528652 # generated CockroachDB client username
+PASSWORD=FlOo0p7jMTXjT27hlZZ-H # generated CockroachDB client password
+DB_NAME=defaultdb
+CLUSTER_NAME=lilac-grizzly-684 # generated CockroachDB client password
+HOST=free-tier21.aws-us-west-2.crdb.io
+TLS_OPTS="sslrootcert=root.crt&sslmode=verify-full&options=--cluster=${CLUSTER_NAME}"
+export CLI_DB_CONNECTION_URL="postgresql://$USER_NAME:$PASSWORD@${HOST}:26257/${DB_NAME}?${TLS_OPTS}"
+cockroach sql --url "${CLI_DB_CONNECTION_URL}" --execute "show tables;"
+~~~
+
+~~~sql
+SHOW TABLES 0
+
+
+Time: 120ms
+~~~
+
+To confirm that the credentials have been properly limited, attempt a forbidden operation. `crdb-role`  does not have permission to list users, so try that in order to generated a permissions error.
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+cockroach sql --url "${CLI_DB_CONNECTION_URL}" --execute "show users;"
+~~~
+
+~~~txt
+ERROR: user v-token-crdb-rol-thflplfwex0k9op0p8qa-1653528652 does not have SELECT privilege on relation users
+SQLSTATE: 42501
+Failed running "sql"
+~~~
```

Comment:
To end the tutorial, are there any other semi-related pages we want to put into a "See also" or "See more" section?